python/pylibcudf/pylibcudf/tests/test_transpose.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import pyarrow as pa
+import pylibcudf as plc
+import pytest
+
+
+@pytest.mark.parametrize(
+    "arr",
+    [
+        [],
+        [1, 2, 3],
+        [1, 2],
+        [1],
+    ],
+)
+def test_transpose(arr):
+    data = {"a": arr, "b": arr}
+    arrow_tbl = pa.table(data)
+    plc_tbl = plc.interop.from_arrow(arrow_tbl)
+    _, plc_result = plc.transpose.transpose(plc_tbl)
+    arrow_result = plc.interop.to_arrow(plc_result)
+    if len(arr) == 0:
+        expected = (0, 0)
+    else:
+        expected = (len(data), len(arr))
+    assert arrow_result.shape == expected
```

Comment:
I don't think there's a way to compute the transpose directly from pyarrow, so something like this might be the next best thing.
```suggestion
@pytest.mark.parametrize(
    "arr",
    [
        [],
        [1, 2, 3],
        [1, 2],
        [1],
        [[0,1],[1,0]],
    ],
)
def test_transpose(arr):
    data = {"a": arr, "b": arr}
    arrow_tbl = pa.table(data)
    plc_tbl = plc.interop.from_arrow(arrow_tbl)
    _, plc_result = plc.transpose.transpose(plc_tbl)
    res = plc.interop.to_arrow(plc_result)
    expect = pa.Table.from_pandas(arrow_tbl.to_pandas().T)
    expect.equals(res)
```

Reply:
Sure thing, good idea. I'll change the comparison to use pandas